rcl/src/rcl/client.c

```diff
@@ -280,6 +280,24 @@ rcl_client_is_valid(const rcl_client_t * client)
     client->impl->rmw_handle, "client's rmw handle is invalid", return false);
   return true;
 }
+
+rcl_ret_t
+rcl_client_set_on_new_response_callback(
+  const rcl_client_t * client,
+  rcl_event_callback_t callback,
+  const void * user_data)
+{
+  if (!rcl_client_is_valid(client)) {
+    // error state already set
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+  RCL_CHECK_FOR_NULL_WITH_MSG(callback, "callback is invalid", return RCL_RET_INVALID_ARGUMENT);
```

Comment:
it should not check if `callback` is NULL or not here? because application tries to unset the callback with `callback == NULL`?